base/interactiveutil.jl

```diff
@@ -254,8 +254,14 @@ function gen_call_with_extracted_types(fcn, ex0)
                     Expr(:call, typesof, map(esc, args[2:end])...))
     end
     exret = Expr(:none)
+    is_macro = false
     ex = expand(ex0)
-    if !isa(ex, Expr)
+    if fcn in [:which, :less, :edit] && ex0.head == :macrocall
+        # special case @edit @time 1+2 to edit the macro
+        is_macro = true
```

Comment:
My main concern is that this doesn't seem to be the way this should work. If it is somehow really necessary to have different behavior for different macros that use this function, it should be controlled explicitly with a optional/keyword argument to this function instead.